### PR TITLE
Refactor ImageSizeControl to functional components.

### PR DESCRIPTION
**Pull request by torounit**
## Description 

Related #22890

Refactored ImageSizeControl to functional components.

Testing instructions

Make sure that uploading images using the Image block "upload" button works as expected

## How has this been tested?

1. Create Image block.
1. Set width, height and reset.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/image-size-control/index.js [hunk 1]
```diff
@@ -13,117 +13,104 @@ import {
...
+	isResizable = true,
+	slug,
+	width,
+	height,
+	onChange,
+	onChangeImage = noop,
+} ) => {
+	const updateDimensions = (
+		nextWidth = undefined,
+		nextHeight = undefined
```
Comment (ZebulanStanphill) [inline packages/block-editor/src/components/image-size-control/index.js:30]: Parameters default to `undefined` anyway, so the default parameters here are unnecessary.

**Review by ItsJonQ — APPROVED** [review]
@torounit Thank you for working on this! The conversion looks good to me. I tested it on my side, and the Image block + it's resizing feature is working as expected.

@ZebulanStanphill 's suggestion regarding `undefined` makes sense. I do understand that it was something that was carried over during the conversion process.

This isn't a blocker for me for merging this PR.

Thanks ❤️

File: packages/block-editor/src/components/image-size-control/index.js (hunk 1 above)
Comment (torounit) [reply packages/block-editor/src/components/image-size-control/index.js]: Thanks Reviewing ! fixed now.

**Comment by ItsJonQ** [comment]
@torounit Awesome! Thank you for that update! We'll wait for Travis to be green and happy. I'll merge it as soon as that's ready

Thank you for contributing 🙏

**Pull request merged**